api/components/BlobDownload.ts

```diff
@@ -0,0 +1,62 @@
+import { ParsedFile } from "@anzp/azure-function-multipart/dist/types/parsed-file.type";
+import { BlobServiceClient } from "@azure/storage-blob";
+require("dotenv").config();
+const blobUrl: string = process.env.REACT_APP_BLOB_STR || undefined;
+if (!blobUrl) {
+  throw new Error("No string attached!");
+}
+
+export interface FileWithMetadata {
+  fileName: {
+    metaData: string
+  }
+
+}
+
+const blobServiceClient = BlobServiceClient.fromConnectionString(blobUrl);
+
+const containers: { name: string; nameShort: string }[] = [];
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const listContainers = async () => {
+  for await (const container of blobServiceClient.listContainers()) {
+    // figure out how to use metadata, for now using names to compare
+    const containerName = container.name;
+    const containerNameShort = containerName.replace(/[^a-z0-9]/gi, "");
+
+    containers.push({ name: containerName, nameShort: containerNameShort });
+  }
+};
+
+const showFiles = async (plot: string) => {
+  let blobData = {};
+  try {
+    await listContainers();
+    const plotReplaced = plot.replace(/[^a-z0-9]/gi, "").toLowerCase();
```

Comment:
We should probably have a centralized function for this, since this transformation (raw plot name to lowercase/removing special characters) is used in multiple places, and it is critical for the upload/download code to match, otherwise it will fail to find the plot name if one of them changes in the future.